Upgrade to zcbor 0.6.0

oyvindronningstad opened the pull request.

oyvindronningstad committed:
lwm2m_senml_cbor_regenerate.sh: Use git am instead of git apply

Since it's better at resolving conflicts.
Commit regenerated files before applying patch.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

oyvindronningstad committed:
lwm2m_senml_cbor_regenerate.sh: Fix argument order in zcbor call

zcbor 0.6.0 has changed the argument order to put code/convert/validate
first.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

oyvindronningstad committed:
lwm2m_senml: Regenerate cbor code using zcbor

Also do clang-format and apply patch.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

oyvindronningstad committed:
west.yml: Update zcbor to 0.6.0

Brings a few quality-of-life improvements for Zephyr, notably:
- No more collision with ARRAY_SIZE()
- C++ improvements

Add entry in release notes.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

oyvindronningstad commented:
cc @VeijoPesonen

nordicjm commented on subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c:
```diff
@@ -49,14 +47,13 @@ static bool encode_repeated_record_bt(zcbor_state_t *state, const struct record_
 	bool tmp_result = ((((zcbor_int32_put(state, (-3)))) &&
-			    ((((*input)._record_bt >= INT64_MIN) &&
-			      ((*input)._record_bt <= INT64_MAX)) ||
+			    ((((*input)._record_bt >= -9223372036854775807LL) &&
+			      ((*input)._record_bt <= 9223372036854775807LL)) ||
```
Any reason for the switch from defines to literals?

oyvindronningstad replied:
```suggestion
			    ((((*input)._record_bt >= INT64_MIN) &&
			      ((*input)._record_bt <= INT64_MAX)) ||
```

VeijoPesonen commented on subsys/net/lib/lwm2m/lwm2m_senml_cbor_types.h:
```diff
@@ -21,9 +21,7 @@
 
 enum lwm2m_senml_cbor_key {
 	lwm2m_senml_cbor_key_bn = -2,
-	lwm2m_senml_cbor_key_bt = -3,
```
Both keys should be put back.

VeijoPesonen reviewed (changes requested):
Array size calculation coming from ZCBOR side is a nice addition.

VeijoPesonen commented:
@jheiskan81 Please review.

oyvindronningstad commented on subsys/net/lib/lwm2m/lwm2m_senml_cbor_decode.c:
```diff
@@ -34,9 +36,24 @@ static bool decode_repeated_record_bn(zcbor_state_t *state, struct record_bn *re
+
+	bool tmp_result = ((((zcbor_int32_expect(state, (-3)))) &&
+			    (zcbor_int64_decode(state, (&(*result)._record_bt))) &&
+			    ((((*result)._record_bt >= -9223372036854775807LL) &&
+			      ((*result)._record_bt <= 9223372036854775807LL)) ||
```
```suggestion
			    ((((*result)._record_bt >= INT64_MIN) &&
			      ((*result)._record_bt <= INT64_MAX)) ||
```

oyvindronningstad commented on subsys/net/lib/lwm2m/lwm2m_senml_cbor_decode.c:
```diff
@@ -48,9 +65,24 @@ static bool decode_repeated_record_n(zcbor_state_t *state, struct record_n *resu
+
+	bool tmp_result = ((((zcbor_uint32_expect(state, (6)))) &&
+			    (zcbor_int64_decode(state, (&(*result)._record_t))) &&
+			    ((((*result)._record_t >= -9223372036854775807LL) &&
+			      ((*result)._record_t <= 9223372036854775807LL)) ||
```
```suggestion
			    ((((*result)._record_t >= INT64_MIN) &&
			      ((*result)._record_t <= INT64_MAX)) ||
```

oyvindronningstad commented on subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c:
```diff
@@ -79,15 +75,14 @@ static bool encode_repeated_record_t(zcbor_state_t *state, const struct record_t
-		   (zcbor_error(state, ZCBOR_ERR_WRONG_RANGE), false)) &&
-		  (zcbor_int64_encode(state, (&(*input)._record_t)))));
+	bool tmp_result = ((((zcbor_uint32_put(state, (6)))) &&
+			    ((((*input)._record_t >= -9223372036854775807LL) &&
+			      ((*input)._record_t <= 9223372036854775807LL)) ||
```
```suggestion
			    ((((*input)._record_t >= INT64_MIN) &&
			      ((*input)._record_t <= INT64_MAX)) ||
```

oyvindronningstad commented on subsys/net/lib/lwm2m/lwm2m_senml_cbor_types.h:
```diff
@@ -21,9 +21,7 @@
 
 enum lwm2m_senml_cbor_key {
 	lwm2m_senml_cbor_key_bn = -2,
```
```suggestion
	lwm2m_senml_cbor_key_bn = -2,
	lwm2m_senml_cbor_key_bt = -3,
```

oyvindronningstad commented on subsys/net/lib/lwm2m/lwm2m_senml_cbor_types.h:
```diff
@@ -21,9 +21,7 @@
 enum lwm2m_senml_cbor_key {
 	lwm2m_senml_cbor_key_bn = -2,
-	lwm2m_senml_cbor_key_bt = -3,
 	lwm2m_senml_cbor_key_n  = 0,
```
```suggestion
	lwm2m_senml_cbor_key_n  = 0,
	lwm2m_senml_cbor_key_t  = 6,
```

oyvindronningstad commented on subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c:
```diff
@@ -139,8 +133,10 @@ static bool encode_value(zcbor_state_t *state, const struct value_ *input)
 				   : (((*input)._value_choice == _value_int)
-					      ? (((((*input)._value_int >= INT64_MIN) &&
-						   ((*input)._value_int <= INT64_MAX)) ||
+					      ? (((((*input)._value_int >=
+						    INT64_MIN) &&
+						   ((*input)._value_int <=
+						    INT64_MAX)) ||
```
```suggestion
					      ? (((((*input)._value_int >= INT64_MIN) &&
						   ((*input)._value_int <= INT64_MAX)) ||
```

rlubos commented:
@oyvindronningstad Out of curiosity, I was under the impression that this code was autogenerated and there's not much room for modifications?

oyvindronningstad committed:
lwm2m_senml_cbor.patch: Update patch file to match current changes

File created by commiting changes and running

  git diff HEAD~1.. > lwm2m_senml_cbor.patch

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

oyvindronningstad commented:
@rlubos The generated code is formatted with clang-format, and then a patch is applied.

jheiskan81 commented on subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c:
```diff
@@ -37,9 +36,8 @@ static bool encode_repeated_record_bn(zcbor_state_t *state, const struct record_
 	bool tmp_result = ((((zcbor_int32_put(state, (-2)))) &&
 			    (zcbor_tstr_encode(state, (&(*input)._record_bn)))));
 
-	if (!tmp_result) {
```
Is this clang rules different than normal zephyr. I was wondering that {
} is mandatory.

oyvindronningstad replied:
I think the clang-format automatically uses the zephyr rules.

VeijoPesonen reviewed (dismissed):
Looks good to me.

nordicjm commented on doc/releases/release-notes-3.3.rst:
```diff
@@ -261,6 +265,19 @@ MCUboot
+changes for Zephyr:
+
+* In the zcbor codebase, the ``ARRAY_SIZE`` macro has been renamed to
+  ``ZCBOR_ARRAY_SIZE`` to not collide with Zephyr's ``ARRAY_SIZE_`` macro.
```
```suggestion
  ``ZCBOR_ARRAY_SIZE`` to not collide with Zephyr's ``ARRAY_SIZE`` macro.
```

oyvindronningstad replied:
Changed to
```
:c:macro:`ARRAY_SIZE`
```

nordicjm commented on west.yml:
```diff
@@ -253,7 +253,7 @@ manifest:
       path: modules/lib/uoscore-uedhoc
     - name: zcbor
-      revision: a0d6981f14d4001d6f0d608d1a427f9bc6bb6d02
+      revision: 0.6.0
```
Should this not be a commit ID?

de-nordic replied:
We have tags now on zcbor fork. The tags are the same as in upstream.

de-nordic commented:
@carlescufi You gave me 1minute to approve ;). I know that I am to slow with reviews, but at least give me a chance :D.
Just kidding: I have already approved yesterday, and there were no changes.
